packages/coinbase/src/index.ts

```diff
@@ -3,51 +3,53 @@ import { WalletInit } from '@web3-onboard/common'
 function coinbaseWallet(options?: { darkMode?: boolean }): WalletInit {
   const { darkMode = false } = options || {}
 
-  return () => {
-    return {
-      label: 'Coinbase',
-      getIcon: async () => (await import('./icon.js')).default,
-      getInterface: async ({ chains, appMetadata }) => {
-        const [chain] = chains
-        const { name, icon } = appMetadata || {}
-
-        const { CoinbaseWalletSDK } = await import('@coinbase/wallet-sdk')
-
-        const base64 = window.btoa(icon || '')
-        const appLogoUrl = `data:image/svg+xml;base64,${base64}`
-
-        const instance = new CoinbaseWalletSDK({
-          appName: name || '',
-          appLogoUrl,
-          darkMode
-        })
-
-        const coinbaseWalletProvider = instance.makeWeb3Provider(
-          chain.rpcUrl,
-          parseInt(chain.id)
-        )
-
-        // patch the chainChanged event
-        const on = coinbaseWalletProvider.on.bind(coinbaseWalletProvider)
-        coinbaseWalletProvider.on = (event, listener) => {
-          on(event, val => {
-            if (event === 'chainChanged') {
-              listener(`0x${(val as number).toString(16)}`)
-              return
+  return ({ device }) => {
+    return device.type === 'mobile'
+      ? {
```

Comment:
Kind of weird to read as a ternary because the return is big but not a blocker

Reply:
yeah good point

Reply:
I will make it more explicit by having the null at the top

Reply:
another approach would be to add a method guard at the top 
```suggestion
      if (device.type === 'mobile') return null
      return {
```
p.s. first time using suggestions button not sure how well it will work lol

Reply:
The guard clause won't work like that as if it is mobile return the wallet otherwise return null -- but I removed some extra fluff so maybe it reads nicer?